storage/range.go

```diff
@@ -169,7 +175,26 @@ func NewRange(desc *proto.RangeDescriptor, rm RangeManager) *Range {
 		respCache:   NewResponseCache(desc.RaftID, rm.Engine()),
 		pendingCmds: map[cmdIDKey]*pendingCmd{},
 	}
-	return r
+
+	// Look in the engine to find the last log index.
+	logKey := engine.RaftLogPrefix(uint64(r.Desc.RaftID))
```

Comment:
Why not move this code to a separate method? Perhaps into engine/keys.go... It's a little chunky for this one.